Remove metadata from Chart widgets

* [x] **I have checked that this bug has not yet been reported by someone else**
* [x] **I have checked that this bug appears on Chrome**
* [x] **I have specified the version** :
- 2024.9.5
* [x] **I have specified my environment** : 
- All

### Stack

```
java.util.concurrent.CompletionException: The field at path '/formEvent' was declared as a non null type, but the code involved in retrieving data has wrongly returned a null value.  The graphql specification requires that the parent field be set to null, or if that is non nullable that it bubble up null to its parent and so on. The non-nullable type is 'FormRefreshedEventPayload' within parent type 'Subscription'
    at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332) ~[na:na]
    at java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:674) 
....
```

and

```
Caused by: graphql.execution.NonNullableFieldWasNullException: The field at path '/formEvent/form/pages[0]/groups[0]/widgets[0]/chart/metadata' was declared as a non null type, but the code involved in retrieving data has wrongly returned a null value.  The graphql specification requires that the parent field be set to null, or if that is non nullable that it bubble up null to its parent and so on. The non-nullable type is 'RepresentationMetadata' within parent type 'PieChart'
    at graphql.execution.NonNullableFieldValidator.validate(NonNullableFieldValidator.java:49) ~[graphql-java-22.1.jar:na]
    at graphql.execution.ExecutionStrategy.completeValueForNull(ExecutionStrategy.java:748) ~[graphql-java-22.1.jar:na]
    ....
```

### Steps to reproduce

Just create a PieChart or a BarChart with the View DSL.

### Expected behavior

PieChart and BarChart can be displayed in forms

### Actual behavior

PieChart and BarChart cannot be displayed in forms

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -35,6 +35,7 @@ The new option ALWAYS allows the separator to be displayed in every case.
 - https://github.com/eclipse-sirius/sirius-web/issues/4050[#4050] [sirius-web] `IUploadFileLoader` must now return an `IResult`.
 This allows an error to be displayed when there is a problem during uploading
 - https://github.com/eclipse-sirius/sirius-web/issues/4037[#4037] [trees] The tree item id passed to `ITreeQueryService.findTreeItem` is no longer a UUID but is now a String.
+- https://github.com/eclipse-sirius/sirius-web/issues/4077[#4077] [charts] Remove `interface RepresentationMetadata` from `BarChart.types.ts`.
 
 === Dependency update
 
@@ -56,6 +57,8 @@ This allows an error to be displayed when there is a problem during uploading
 - https://github.com/eclipse-sirius/sirius-web/issues/4051[#4051] Error when deleting Resource that use EReference.isContainer feature
 - https://github.com/eclipse-sirius/sirius-web/issues/3958[#3958] [trees] Prevent expand all action to loop indefinitely.
 - https://github.com/eclipse-sirius/sirius-web/issues/4073[#4073] [trees] Add missing variables `ancestorIds` and `index` to variable manager used in the default expand all handler.
+- https://github.com/eclipse-sirius/sirius-web/issues/4077[#4077] [charts] Remove metadata from Charts widgets.
+Charts' metadata is only relevant when Charts are used as Representations.
 
 === New Features
 

--- a/packages/charts/backend/sirius-components-charts/src/main/java/org/eclipse/sirius/components/charts/barchart/BarChart.java
+++ b/packages/charts/backend/sirius-components-charts/src/main/java/org/eclipse/sirius/components/charts/barchart/BarChart.java
@@ -48,6 +48,7 @@ public final class BarChart implements IChart {
 
     private int height;
 
+    private String yAxisLabel;
 
     private BarChart() {
         // Prevent instantiation
@@ -94,14 +95,18 @@ public final class BarChart implements IChart {
         return this.height;
     }
 
+    public String getYAxisLabel() {
+        return this.yAxisLabel;
+    }
+
     public static Builder newBarChart(String id) {
         return new Builder(id);
     }
 
     @Override
     public String toString() {
-        String pattern = "{0} '{'id: {1}, descriptionId: {2}, label: {3}, kind: {4}, width: {5}, height: {6}'}'";
-        return MessageFormat.format(pattern, this.getClass().getSimpleName(), this.id, this.descriptionId, this.label, this.kind, this.width, this.height);
+        String pattern = "{0} '{'id: {1}, descriptionId: {2}, label: {3}, kind: {4}, width: {5}, height: {6}, yAxisLabel:{7}'}'";
+        return MessageFormat.format(pattern, this.getClass().getSimpleName(), this.id, this.descriptionId, this.label, this.kind, this.width, this.height, this.yAxisLabel);
     }
 
     /**
@@ -129,6 +134,8 @@ public final class BarChart implements IChart {
         private int width;
 
         private int height;
+
+        private String yAxisLabel;
 
         public Builder(String id) {
             this.id = Objects.requireNonNull(id);
@@ -169,6 +176,11 @@ public final class BarChart implements IChart {
             return this;
         }
 
+        public Builder yAxisLabel(String yAxisLabel) {
+            this.yAxisLabel = Objects.requireNonNull(yAxisLabel);
+            return this;
+        }
+
         public BarChart build() {
             BarChart barChart = new BarChart();
             barChart.id = Objects.requireNonNull(this.id);
@@ -177,9 +189,10 @@ public final class BarChart implements IChart {
             barChart.kind = Objects.requireNonNull(this.kind);
             barChart.targetObjectId = Objects.requireNonNull(this.targetObjectId);
             barChart.entries = Objects.requireNonNull(this.entries);
-            barChart.style = this.style;
+            barChart.style = this.style; // Optional on purpose
             barChart.width = this.width;
             barChart.height = this.height;
+            barChart.yAxisLabel = this.yAxisLabel; // Optional on purpose
             return barChart;
         }
     }

--- a/packages/charts/backend/sirius-components-charts/src/main/java/org/eclipse/sirius/components/charts/barchart/components/BarChartComponent.java
+++ b/packages/charts/backend/sirius-components-charts/src/main/java/org/eclipse/sirius/components/charts/barchart/components/BarChartComponent.java
@@ -49,9 +49,9 @@ public class BarChartComponent implements IComponent {
         String targetObjectId = barChartDescription.getTargetObjectIdProvider().apply(variableManager);
         List<Number> values = barChartDescription.getValuesProvider().apply(variableManager);
         List<String> keys = barChartDescription.getKeysProvider().apply(variableManager);
+        String yAxisLabel = barChartDescription.getYAxisLabelProvider().apply(variableManager);
         BarChartStyle barChartStyle = barChartDescription.getStyleProvider().apply(variableManager);
 
-        // @formatter:off
         Builder builder = BarChartElementProps.newBarChartElementProps(id)
                 .label(label)
                 .descriptionId(barChartDescription.getId())
@@ -60,10 +60,12 @@ public class BarChartComponent implements IComponent {
                 .width(barChartDescription.getWidth())
                 .height(barChartDescription.getHeight())
                 .keys(keys);
-        // @formatter:on
 
         if (barChartStyle != null) {
             builder.style(barChartStyle);
+        }
+        if (yAxisLabel != null) {
+            builder.yAxisLabel(yAxisLabel);
         }
         BarChartElementProps barChartElementProps = builder.build();
         return new Element(BarChartElementProps.TYPE, barChartElementProps);

--- a/packages/charts/backend/sirius-components-charts/src/main/java/org/eclipse/sirius/components/charts/barchart/descriptions/BarChartDescription.java
+++ b/packages/charts/backend/sirius-components-charts/src/main/java/org/eclipse/sirius/components/charts/barchart/descriptions/BarChartDescription.java
@@ -47,6 +47,8 @@ public final class BarChartDescription implements IChartDescription {
 
     private int height;
 
+    private Function<VariableManager, String> yAxisLabelProvider;
+
     private BarChartDescription() {
         // prevent instantiation
     }
@@ -89,6 +91,10 @@ public final class BarChartDescription implements IChartDescription {
         return this.height;
     }
 
+    public Function<VariableManager, String> getYAxisLabelProvider() {
+        return this.yAxisLabelProvider;
+    }
+
     public static Builder newBarChartDescription(String id) {
         return new Builder(id);
     }
@@ -117,6 +123,8 @@ public final class BarChartDescription implements IChartDescription {
         private int width;
 
         private int height;
+
+        private Function<VariableManager, String> yAxisLabelProvider = variableManager -> null;
 
         public Builder(String id) {
             this.id = Objects.requireNonNull(id);
@@ -162,6 +170,11 @@ public final class BarChartDescription implements IChartDescription {
             return this;
         }
 
+        public Builder yAxisLabelProvider(Function<VariableManager, String> yAxisLabelProvider) {
+            this.yAxisLabelProvider = Objects.requireNonNull(yAxisLabelProvider);
+            return this;
+        }
+
         public BarChartDescription build() {
             BarChartDescription barChartDescription = new BarChartDescription();
             barChartDescription.id = Objects.requireNonNull(this.id);
@@ -173,6 +186,7 @@ public final class BarChartDescription implements IChartDescription {
             barChartDescription.styleProvider = Objects.requireNonNull(this.styleProvider);
             barChartDescription.width = this.width;
             barChartDescription.height = this.height;
+            barChartDescription.yAxisLabelProvider = Objects.requireNonNull(this.yAxisLabelProvider);
             return barChartDescription;
         }
     }

--- a/packages/charts/backend/sirius-components-charts/src/main/java/org/eclipse/sirius/components/charts/barchart/elements/BarChartElementProps.java
+++ b/packages/charts/backend/sirius-components-charts/src/main/java/org/eclipse/sirius/components/charts/barchart/elements/BarChartElementProps.java
@@ -47,6 +47,8 @@ public final class BarChartElementProps implements IProps {
 
     private int height;
 
+    private String yAxisLabel;
+
     private BarChartElementProps() {
         // prevent instantiation
     }
@@ -87,6 +89,10 @@ public final class BarChartElementProps implements IProps {
         return this.height;
     }
 
+    public String getYAxisLabel() {
+        return this.yAxisLabel;
+    }
+
     public static Builder newBarChartElementProps(String id) {
         return new Builder(id);
     }
@@ -115,6 +121,8 @@ public final class BarChartElementProps implements IProps {
         private int width;
 
         private int height;
+
+        private String yAxisLabel;
 
         public Builder(String id) {
             this.id = Objects.requireNonNull(id);
@@ -160,6 +168,11 @@ public final class BarChartElementProps implements IProps {
             return this;
         }
 
+        public Builder yAxisLabel(String yAxisLabel) {
+            this.yAxisLabel = Objects.requireNonNull(yAxisLabel);
+            return this;
+        }
+
         public BarChartElementProps build() {
             BarChartElementProps barChartElementProps = new BarChartElementProps();
             barChartElementProps.id = Objects.requireNonNull(this.id);
@@ -171,6 +184,7 @@ public final class BarChartElementProps implements IProps {
             barChartElementProps.style = this.style; // Optional on purpose
             barChartElementProps.width = this.width;
             barChartElementProps.height = this.height;
+            barChartElementProps.yAxisLabel = this.yAxisLabel; // Optional on purpose
             return barChartElementProps;
         }
     }

--- a/packages/charts/backend/sirius-components-collaborative-charts/src/main/resources/schema/charts.graphqls
+++ b/packages/charts/backend/sirius-components-collaborative-charts/src/main/resources/schema/charts.graphqls
@@ -30,6 +30,7 @@ type BarChart implements Representation {
   style: BarChartStyle
   width: Int!
   height: Int!
+  yAxisLabel: String
 }
 
 type BarChartEntry {

--- a/packages/charts/frontend/sirius-components-charts/src/bar/BarChart.tsx
+++ b/packages/charts/frontend/sirius-components-charts/src/bar/BarChart.tsx
@@ -35,11 +35,7 @@ export const BarChart = ({ chart }: BarChartProps) => {
     const yRange = [height - marginBottom, marginTop]; // [bottom, top]
 
     if (d3Container.current && chart) {
-      const {
-        entries: data,
-        metadata: { label: yLabel },
-        style,
-      } = chart;
+      const { entries: data, yAxisLabel, style } = chart;
       const fontSize = getFontSize(style);
       const fontStyle = getFontStyle(style);
       const fontWeight = getFontWeight(style);
@@ -98,7 +94,7 @@ export const BarChart = ({ chart }: BarChartProps) => {
             .attr('y', 10)
             .attr('fill', 'currentColor')
             .attr('text-anchor', 'start')
-            .text(yLabel)
+            .text(yAxisLabel)
         );
 
       const bar = svg

--- a/packages/charts/frontend/sirius-components-charts/src/bar/BarChart.types.ts
+++ b/packages/charts/frontend/sirius-components-charts/src/bar/BarChart.types.ts
@@ -18,15 +18,11 @@ export interface BarChartProps {
 }
 
 export interface BarChartRepresentation {
-  metadata: RepresentationMetadata;
+  yAxisLabel: string | null;
   entries: BarChartRepresentationEntry[];
   style: BarChartStyle | null;
   width: number;
   height: number;
-}
-
-export interface RepresentationMetadata {
-  label: string;
 }
 
 export interface BarChartRepresentationEntry {

--- a/packages/formdescriptioneditors/backend/sirius-components-formdescriptioneditors/src/main/java/org/eclipse/sirius/components/formdescriptioneditors/components/ViewFormDescriptionEditorConverterSwitch.java
+++ b/packages/formdescriptioneditors/backend/sirius-components-formdescriptioneditors/src/main/java/org/eclipse/sirius/components/formdescriptioneditors/components/ViewFormDescriptionEditorConverterSwitch.java
@@ -563,7 +563,6 @@ public class ViewFormDescriptionEditorConverterSwitch extends FormSwitch<Abstrac
             return new BarChartStyleProvider(style).build();
         };
 
-        // @formatter:off
         IChartDescription chartDescription = BarChartDescription.newBarChartDescription(UUID.randomUUID().toString())
                 .label(Optional.ofNullable(viewBarChartDescription.getName()).orElse(""))
                 .labelProvider(vm -> this.getWidgetLabel(viewBarChartDescription, "BarChart"))
@@ -573,8 +572,9 @@ public class ViewFormDescriptionEditorConverterSwitch extends FormSwitch<Abstrac
                 .styleProvider(styleProvider)
                 .width(viewBarChartDescription.getWidth())
                 .height(viewBarChartDescription.getHeight())
+                .yAxisLabelProvider(vm -> this.getYAxisLabel(viewBarChartDescription, "Y Axis Label"))
                 .build();
-        // @formatter:on
+
         return this.createChartWidgetDescription(viewBarChartDescription, chartDescription);
     }
 
@@ -588,7 +588,6 @@ public class ViewFormDescriptionEditorConverterSwitch extends FormSwitch<Abstrac
             return new PieChartStyleProvider(style).build();
         };
 
-        // @formatter:off
         IChartDescription chartDescription =  PieChartDescription.newPieChartDescription(UUID.randomUUID().toString())
                 .label(this.getWidgetLabel(viewPieChartDescription, "PieChart"))
                 .targetObjectIdProvider(vm -> vm.get(IEditingContext.EDITING_CONTEXT, IEditingContext.class).map(IEditingContext::getId).orElse(null))
@@ -596,7 +595,7 @@ public class ViewFormDescriptionEditorConverterSwitch extends FormSwitch<Abstrac
                 .valuesProvider(vm -> List.of())
                 .styleProvider(styleProvider)
                 .build();
-        // @formatter:on
+
         return this.createChartWidgetDescription(viewPieChartDescription, chartDescription);
     }
 
@@ -763,4 +762,15 @@ public class ViewFormDescriptionEditorConverterSwitch extends FormSwitch<Abstrac
         return helpText;
     }
 
+    public String getYAxisLabel(org.eclipse.sirius.components.view.form.BarChartDescription barChartDescription, String defaultLabel) {
+        String yAxisLabel = defaultLabel;
+        String name = barChartDescription.getName();
+        String labelExpression = barChartDescription.getYAxisLabelExpression();
+        if (labelExpression != null && !labelExpression.isBlank() && !labelExpression.startsWith(AQL_PREFIX)) {
+            yAxisLabel = labelExpression;
+        } else if (name != null && !name.isBlank()) {
+            yAxisLabel = name;
+        }
+        return yAxisLabel;
+    }
 }

--- a/packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/BarChartWidget.tsx
+++ b/packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/BarChartWidget.tsx
@@ -38,9 +38,6 @@ export const BarChartWidget = ({ widget }: BarChartWidgetProps) => {
   const barChartWidget = widget.chart as GQLBarChart;
 
   const chart: BarChartRepresentation = {
-    metadata: {
-      label: 'Frequency',
-    },
     entries: [
       { key: 'A', value: 0.08167 },
       { key: 'B', value: 0.01492 },
@@ -72,6 +69,7 @@ export const BarChartWidget = ({ widget }: BarChartWidgetProps) => {
     style: barChartWidget.style,
     width: barChartWidget.width,
     height: barChartWidget.height,
+    yAxisLabel: 'Frequency',
   };
   const [selected, setSelected] = useState<boolean>(false);
   const { selection } = useSelection();
@@ -89,7 +87,7 @@ export const BarChartWidget = ({ widget }: BarChartWidgetProps) => {
 
   return (
     <div
-      data-testid={barChartWidget.metadata.label}
+      data-testid={widget.label}
       onFocus={() => setSelected(true)}
       onBlur={() => setSelected(false)}
       ref={ref}

--- a/packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/PieChartWidget.tsx
+++ b/packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/PieChartWidget.tsx
@@ -15,8 +15,8 @@ import { useSelection } from '@eclipse-sirius/sirius-components-core';
 import { GQLPieChart } from '@eclipse-sirius/sirius-components-forms';
 import HelpOutlineOutlined from '@mui/icons-material/HelpOutlineOutlined';
 import Typography from '@mui/material/Typography';
-import { makeStyles } from 'tss-react/mui';
 import { useEffect, useRef, useState } from 'react';
+import { makeStyles } from 'tss-react/mui';
 import { PieChartWidgetProps } from './WidgetEntry.types';
 
 const useStyles = makeStyles()((theme) => ({
@@ -72,14 +72,14 @@ export const PieChartWidget = ({ widget }: PieChartWidgetProps) => {
 
   return (
     <div
-      data-testid={pieChartWidget.metadata.label}
+      data-testid={widget.label}
       onFocus={() => setSelected(true)}
       onBlur={() => setSelected(false)}
       ref={ref}
       tabIndex={0}>
       <div className={classes.propertySectionLabel}>
         <Typography variant="subtitle2" className={selected ? classes.selected : ''}>
-          {pieChartWidget.metadata.label}
+          {widget.label}
         </Typography>
         {widget.hasHelpText ? <HelpOutlineOutlined color="secondary" style={{ marginLeft: 8, fontSize: 16 }} /> : null}
       </div>

--- a/packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java
+++ b/packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java
@@ -154,15 +154,14 @@ public class FormElementFactory implements IElementFactory {
 
     private Object instantiateBarChart(BarChartElementProps props) {
         List<BarChartEntry> entries = this.getBarChartEntries(props);
-        // @formatter:off
         Builder builder = BarChart.newBarChart(props.getId())
                 .descriptionId(props.getDescriptionId())
                 .label(props.getLabel())
                 .targetObjectId(props.getTargetObjectId())
                 .width(props.getWidth())
                 .height(props.getHeight())
-                .entries(entries);
-        // @formatter:on
+                .entries(entries)
+                .yAxisLabel(props.getYAxisLabel());
 
         if (props.getStyle() != null) {
             builder.style(props.getStyle());
@@ -173,13 +172,11 @@ public class FormElementFactory implements IElementFactory {
 
     private Object instantiatePieChart(PieChartElementProps props) {
         List<PieChartEntry> entries = this.getPieChartEntries(props);
-        // @formatter:off
         org.eclipse.sirius.components.charts.piechart.PieChart.Builder builder = PieChart.newPieChart(props.getId())
                 .descriptionId(props.getDescriptionId())
                 .targetObjectId(props.getTargetObjectId())
                 .label(props.getLabel())
                 .entries(entries);
-        // @formatter:on
 
         if (props.getStyle() != null) {
             builder.style(props.getStyle());
@@ -213,7 +210,6 @@ public class FormElementFactory implements IElementFactory {
     }
 
     private Form instantiateForm(FormElementProps props, List<Object> children) {
-        // @formatter:off
         List<Page> pages = children.stream()
                 .filter(Page.class::isInstance)
                 .map(Page.class::cast)
@@ -225,7 +221,6 @@ public class FormElementFactory implements IElementFactory {
                 .descriptionId(props.getDescriptionId())
                 .pages(pages)
                 .build();
-        // @formatter:on
     }
 
     private Page instantiatePage(PageElementProps props, List<Object> children) {
@@ -269,7 +264,6 @@ public class FormElementFactory implements IElementFactory {
     private Checkbox instantiateCheckbox(CheckboxElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         Checkbox.Builder checkboxBuilder = Checkbox.newCheckbox(props.getId())
                 .label(props.getLabel())
                 .value(props.isValue())
@@ -287,19 +281,17 @@ public class FormElementFactory implements IElementFactory {
             checkboxBuilder.helpTextProvider(props.getHelpTextProvider());
         }
         return checkboxBuilder.build();
-        // @formatter:on
     }
 
     private org.eclipse.sirius.components.forms.List instantiateList(ListElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         org.eclipse.sirius.components.forms.List.Builder listBuilder = org.eclipse.sirius.components.forms.List.newList(props.getId())
                 .label(props.getLabel())
                 .items(props.getItems())
                 .diagnostics(diagnostics)
                 .readOnly(props.isReadOnly());
-        // @formatter:on
+
         if (props.getIconURL() != null) {
             listBuilder.iconURL(props.getIconURL());
         }
@@ -315,7 +307,6 @@ public class FormElementFactory implements IElementFactory {
     private Radio instantiateRadio(RadioElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         Radio.Builder radioBuilder = Radio.newRadio(props.getId())
                 .label(props.getLabel())
                 .options(props.getOptions())
@@ -334,13 +325,11 @@ public class FormElementFactory implements IElementFactory {
         }
 
         return radioBuilder.build();
-        // @formatter:on
     }
 
     private Select instantiateSelect(SelectElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         Select.Builder selectBuilder = Select.newSelect(props.getId())
                 .label(props.getLabel())
                 .options(props.getOptions())
@@ -359,13 +348,11 @@ public class FormElementFactory implements IElementFactory {
             selectBuilder.helpTextProvider(props.getHelpTextProvider());
         }
         return selectBuilder.build();
-        // @formatter:on
     }
 
     private MultiSelect instantiateMultiSelect(MultiSelectElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         MultiSelect.Builder multiSelectBuilder = MultiSelect.newMultiSelect(props.getId())
                 .label(props.getLabel())
                 .options(props.getOptions())
@@ -384,13 +371,11 @@ public class FormElementFactory implements IElementFactory {
             multiSelectBuilder.helpTextProvider(props.getHelpTextProvider());
         }
         return multiSelectBuilder.build();
-        // @formatter:on
     }
 
     private Textarea instantiateTextarea(TextareaElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         Textarea.Builder textareaBuilder = Textarea.newTextarea(props.getId())
                 .label(props.getLabel())
                 .value(props.getValue())
@@ -411,20 +396,17 @@ public class FormElementFactory implements IElementFactory {
             textareaBuilder.helpTextProvider(props.getHelpTextProvider());
         }
         return textareaBuilder.build();
-        // @formatter:on
     }
 
     private Textfield instantiateTextfield(TextfieldElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         Textfield.Builder textfieldBuilder = Textfield.newTextfield(props.getId())
                 .label(props.getLabel())
                 .value(props.getValue())
                 .newValueHandler(props.getNewValueHandler())
                 .diagnostics(diagnostics)
                 .readOnly(props.isReadOnly());
-        // @formatter:on
 
         if (props.getCompletionProposalsProvider() != null) {
             textfieldBuilder.completionProposalsProvider(props.getCompletionProposalsProvider());
@@ -442,23 +424,19 @@ public class FormElementFactory implements IElementFactory {
     }
 
     private Object instantiateDiagnostic(DiagnosticElementProps props, List<Object> children) {
-        // @formatter:off
         return Diagnostic.newDiagnostic(props.getId())
                 .kind(props.getKind())
                 .message(props.getMessage())
                 .build();
-        // @formatter:on
     }
 
     private Link instantiateLink(LinkElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         Link.Builder linkbuilder = Link.newLink(props.getId())
                 .label(props.getLabel())
                 .url(props.getUrl())
                 .diagnostics(diagnostics);
-        // @formatter:on
 
         if (props.getStyle() != null) {
             linkbuilder.style(props.getStyle());
@@ -475,13 +453,12 @@ public class FormElementFactory implements IElementFactory {
     private Button instantiateButton(ButtonElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         Button.Builder buttonBuilder = Button.newButton(props.getId())
                 .label(props.getLabel())
                 .pushButtonHandler(props.getPushButtonHandler())
                 .diagnostics(diagnostics)
                 .readOnly(props.isReadOnly());
-        // @formatter:on
+
         if (props.getIconURL() != null) {
             buttonBuilder.iconURL(props.getIconURL());
         }
@@ -523,12 +500,11 @@ public class FormElementFactory implements IElementFactory {
     private ToolbarAction instantiateToolbarAction(ToolbarActionElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         ToolbarAction.Builder buttonBuilder = ToolbarAction.newToolbarAction(props.getId())
                 .label(props.getLabel())
                 .pushButtonHandler(props.getPushButtonHandler())
                 .diagnostics(diagnostics);
-        // @formatter:on
+
         if (props.getIconURL() != null) {
             buttonBuilder.iconURL(props.getIconURL());
         }
@@ -590,7 +566,6 @@ public class FormElementFactory implements IElementFactory {
     private LabelWidget instantiateLabel(LabelWidgetElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         LabelWidget.Builder labelBuilder = LabelWidget.newLabelWidget(props.getId())
                 .label(props.getLabel())
                 .value(props.getValue())
@@ -603,12 +578,11 @@ public class FormElementFactory implements IElementFactory {
             labelBuilder.helpTextProvider(props.getHelpTextProvider());
         }
         return labelBuilder.build();
-        // @formatter:on
     }
 
     private Object instantiateChartWidget(ChartWidgetElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
-        // @formatter:off
+
         var chart = children.stream()
                 .filter(IChart.class::isInstance)
                 .map(IChart.class::cast)
@@ -619,7 +593,7 @@ public class FormElementFactory implements IElementFactory {
                 .label(props.getLabel())
                 .chart(chart)
                 .diagnostics(diagnostics);
-        // @formatter:on
+
         if (props.getIconURL() != null) {
             chartBuilder.iconURL(props.getIconURL());
         }
@@ -660,7 +634,6 @@ public class FormElementFactory implements IElementFactory {
     private TreeWidget instantiateTree(TreeElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         TreeWidget.Builder builder = TreeWidget.newTreeWidget(props.getId())
                 .label(props.getLabel())
                 .iconURL(props.getIconURL())
@@ -668,7 +641,7 @@ public class FormElementFactory implements IElementFactory {
                 .expandedNodesIds(props.getExpandedNodesIds())
                 .readOnly(props.isReadOnly())
                 .diagnostics(diagnostics);
-        // @formatter:on
+
         if (props.getHelpTextProvider() != null) {
             builder.helpTextProvider(props.getHelpTextProvider());
         }
@@ -678,12 +651,10 @@ public class FormElementFactory implements IElementFactory {
     private Image instantiateImage(ImageElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         Image.Builder imagebuilder = Image.newImage(props.getId())
                 .label(props.getLabel())
                 .url(props.getUrl())
                 .diagnostics(diagnostics);
-        // @formatter:on
 
         if (props.getIconURL() != null) {
             imagebuilder.iconURL(props.getIconURL());
@@ -700,7 +671,6 @@ public class FormElementFactory implements IElementFactory {
     private RichText instantiateRichText(RichTextElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
-        // @formatter:off
         RichText.Builder builder = RichText.newRichText(props.getId())
                 .label(props.getLabel())
                 .value(props.getValue())
@@ -714,16 +684,13 @@ public class FormElementFactory implements IElementFactory {
             builder.helpTextProvider(props.getHelpTextProvider());
         }
         return builder.build();
-        // @formatter:on
     }
 
     private List<Diagnostic> getDiagnosticsFromChildren(List<Object> children) {
-        // @formatter:off
         return children.stream()
                 .filter(Diagnostic.class::isInstance)
                 .map(Diagnostic.class::cast)
                 .toList();
-        // @formatter:on
     }
 
 }

--- a/packages/forms/frontend/sirius-components-forms/src/form/FormEventFragments.ts
+++ b/packages/forms/frontend/sirius-components-forms/src/form/FormEventFragments.ts
@@ -205,9 +205,6 @@ export const widgetFields = `
     chart {
       __typename
       ... on BarChart {
-        metadata {
-          label
-        }
         entries {
           key
           value
@@ -222,11 +219,9 @@ export const widgetFields = `
         }
         width
         height
+        yAxisLabel
       }
       ... on PieChart {
-        metadata {
-          label
-        }
         entries {
           key
           value

--- a/packages/forms/frontend/sirius-components-forms/src/form/FormEventFragments.types.ts
+++ b/packages/forms/frontend/sirius-components-forms/src/form/FormEventFragments.types.ts
@@ -276,17 +276,13 @@ export interface GQLChartWidget extends GQLWidget {
 
 export type GQLChart = GQLBarChart | GQLPieChart;
 
-export interface GQLChartMetadata {
-  label: string;
-}
-
 export interface GQLBarChart {
   __typename: string;
-  metadata: GQLChartMetadata;
   entries: GQLBarChartEntry[];
   style: GQLBarChartStyle | null;
   width: number;
   height: number;
+  yAxisLabel: string | null;
 }
 
 export interface GQLFontStyle {
@@ -308,7 +304,6 @@ export interface GQLBarChartEntry {
 
 export interface GQLPieChart {
   __typename: string;
-  metadata: GQLChartMetadata;
   entries: GQLPieChartEntry[];
   style: GQLPieChartStyle | null;
 }

--- a/packages/forms/frontend/sirius-components-forms/src/propertysections/__tests__/ChartWidgetPropertySection.test.tsx
+++ b/packages/forms/frontend/sirius-components-forms/src/propertysections/__tests__/ChartWidgetPropertySection.test.tsx
@@ -11,13 +11,12 @@
  *     Obeo - initial API and implementation
  *******************************************************************************/
 import { MockedProvider } from '@apollo/client/testing';
-
 import { cleanup, render } from '@testing-library/react';
+import React from 'react';
 import { afterEach, expect, test } from 'vitest';
 import {
   GQLBarChart,
   GQLBarChartEntry,
-  GQLChartMetadata,
   GQLChartWidget,
   GQLPieChart,
   GQLPieChartEntry,
@@ -25,6 +24,7 @@ import {
 import { ChartWidgetPropertySection } from '../ChartWidgetPropertySection';
 
 afterEach(() => cleanup());
+
 const pieChartentries: GQLPieChartEntry[] = [
   { key: 'a', value: 10 },
   { key: 'b', value: 15 },
@@ -32,20 +32,14 @@ const pieChartentries: GQLPieChartEntry[] = [
   { key: 'd', value: 20 },
 ];
 
-const pieChartMetadata: GQLChartMetadata = {
-  label: 'PieChartLabel',
-};
-
 const defaultPieChart: GQLPieChart = {
   __typename: 'PieChart',
-  metadata: pieChartMetadata,
   entries: pieChartentries,
   style: null,
 };
 
 const defaultPieChartWithStyle: GQLPieChart = {
   __typename: 'PieChart',
-  metadata: pieChartMetadata,
   entries: pieChartentries,
   style: {
     strokeWidth: 2,
@@ -61,7 +55,6 @@ const defaultPieChartWithStyle: GQLPieChart = {
 
 const defaultPieChartWithEmptyStyle: GQLPieChart = {
   __typename: 'PieChart',
-  metadata: pieChartMetadata,
   entries: pieChartentries,
   style: {
     strokeWidth: null,
@@ -81,20 +74,17 @@ const barChartentries: GQLBarChartEntry[] = [
   { key: 'c', value: 5 },
   { key: 'd', value: 20 },
 ];
-const barChartMetadata: GQLChartMetadata = {
-  label: 'values',
-};
+
 const defaultBarChart: GQLBarChart = {
   __typename: 'BarChart',
-  metadata: barChartMetadata,
   entries: barChartentries,
   style: null,
   width: 500,
   height: 250,
+  yAxisLabel: 'values',
 };
 const defaultBarChartWithStyle: GQLBarChart = {
   __typename: 'BarChart',
-  metadata: barChartMetadata,
   entries: barChartentries,
   style: {
     barsColor: 'Aquamarine',
@@ -106,11 +96,11 @@ const defaultBarChartWithStyle: GQLBarChart = {
   },
   width: 500,
   height: 250,
+  yAxisLabel: 'values',
 };
 
 const defaultBarChartWithEmptyStyle: GQLBarChart = {
   __typename: 'BarChart',
-  metadata: barChartMetadata,
   entries: barChartentries,
   style: {
     barsColor: '',
@@ -122,6 +112,7 @@ const defaultBarChartWithEmptyStyle: GQLBarChart = {
   },
   width: 500,
   height: 250,
+  yAxisLabel: 'values',
 };
 const defaultPieChartWidget: GQLChartWidget = {
   id: 'id',

--- a/packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/form/ViewFormDescriptionConverterSwitch.java
+++ b/packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/form/ViewFormDescriptionConverterSwitch.java
@@ -435,9 +435,10 @@ public class ViewFormDescriptionConverterSwitch extends FormSwitch<Optional<Abst
 
     @Override
     public Optional<AbstractControlDescription> caseBarChartDescription(org.eclipse.sirius.components.view.form.BarChartDescription viewBarChartDescription) {
-        String labelExpression = viewBarChartDescription.getYAxisLabelExpression();
+        String labelExpression = viewBarChartDescription.getLabelExpression();
         String keysExpression = viewBarChartDescription.getKeysExpression();
         String valuesExpression = viewBarChartDescription.getValuesExpression();
+        String yAxisLabelExpression = viewBarChartDescription.getYAxisLabelExpression();
         Function<VariableManager, BarChartStyle> styleProvider = new BarChartStyleProvider(this.interpreter, viewBarChartDescription);
 
         if (labelExpression == null || keysExpression == null || valuesExpression == null) {
@@ -452,7 +453,8 @@ public class ViewFormDescriptionConverterSwitch extends FormSwitch<Optional<Abst
                 .valuesProvider(this.getMultiValueProvider(valuesExpression, Number.class))
                 .styleProvider(styleProvider)
                 .width(viewBarChartDescription.getWidth())
-                .height(viewBarChartDescription.getHeight());
+                .height(viewBarChartDescription.getHeight())
+                .yAxisLabelProvider(this.getStringValueProvider(yAxisLabelExpression));
         return Optional.of(this.createChartWidgetDescription(viewBarChartDescription, builder.build()));
     }
 

--- a/packages/view/backend/sirius-components-view-emf/src/test/java/org/eclipse/sirius/components/view/emf/view/DynamicFormsTests.java
+++ b/packages/view/backend/sirius-components-view-emf/src/test/java/org/eclipse/sirius/components/view/emf/view/DynamicFormsTests.java
@@ -474,7 +474,7 @@ public class DynamicFormsTests {
         IChart chart = chartWidgetWithBarChart.getChart();
         assertThat(chart).isInstanceOf(BarChart.class);
         BarChart barChart = (BarChart) chart;
-        assertThat(barChart.getLabel()).isEqualTo("the values");
+        assertThat(barChart.getYAxisLabel()).isEqualTo("the values");
         List<BarChartEntry> barChartEntries = barChart.getEntries();
         this.checkBarChartEntry(barChartEntries, 0, "a", 1);
         this.checkBarChartEntry(barChartEntries, 1, "b", 3);

